File: PorizoAndroid/Sources/PorizoSkipSpike/PoemLibrary.swift
import Foundation

/// Pure library logic for the Poems tab (U6), mirroring SongLibrary.

enum PoemLibraryFilter: String, CaseIterable, Identifiable {
    case mine
    case received

    var id: String { rawValue }
    var label: String { self == .mine ? "My Poems" : "Received" }
}

enum PoemLibrary {

    /// Filter poems by the My/Received segmented control.
    static func filtered(_ poems: [PorizoPoemSummary], by filter: PoemLibraryFilter) -> [PorizoPoemSummary] {
        switch filter {
        case .mine: return poems.filter { !$0.isReceived }
        case .received: return poems.filter { $0.isReceived }
        }
    }

    /// A one-line preview drawn from the first non-empty verse.
    static func preview(for poem: PorizoPoemSummary) -> String {
        poem.verses.first(where: { !$0.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty })
            ?? "Tap to read"
    }

    /// Build a playable track for the U3 player from a poem's TTS audio URL.
    static func playableTrack(for poem: PorizoPoemSummary, audioURL: String) -> AndroidPlayableTrack {
        AndroidPlayableTrack(
            id: poem.id,
            title: poem.title,
            recipientName: poem.recipientName,
            artworkURL: nil,
            streamURL: audioURL,
            isOwnedContent: !poem.isReceived
        )
    }
}

File: PorizoAndroid/Android/core/domain/src/main/kotlin/com/porizo/core/domain/library/PoemLibrary.kt
package com.porizo.core.domain.library

import com.porizo.core.model.PlayableTrack
import com.porizo.core.model.PoemSummary

enum class PoemLibraryFilter(val label: String) {
    Mine("My Poems"),
    Received("Received"),
}

object PoemLibrary {
    fun filtered(poems: List<PoemSummary>, filter: PoemLibraryFilter): List<PoemSummary> =
        when (filter) {
            PoemLibraryFilter.Mine -> poems.filterNot { it.isReceived }
            PoemLibraryFilter.Received -> poems.filter { it.isReceived }
        }

    fun preview(poem: PoemSummary): String =
        poem.verses.firstOrNull { it.trim().isNotEmpty() } ?: "Tap to read"

    fun playableTrack(poem: PoemSummary, audioUrl: String): PlayableTrack =
        PlayableTrack(
            id = poem.id,
            title = poem.title,
            recipientName = poem.recipientName,
            artworkUrl = null,
            streamUrl = audioUrl,
            isOwnedContent = !poem.isReceived,
        )
}
